test/Distributed/Macros/distributed_macro_expansion_DistributedProtocol_various_requirements.swift

```diff
@@ -0,0 +1,98 @@
+// REQUIRES: swift_swift_parser, asserts
+//
+// UNSUPPORTED: back_deploy_concurrency
+// REQUIRES: concurrency
+// REQUIRES: distributed
+//
+// RUN: %empty-directory(%t)
+// RUN: %empty-directory(%t-scratch)
+
+// RUN: %target-swift-frontend-emit-module -emit-module-path %t/FakeDistributedActorSystems.swiftmodule -module-name FakeDistributedActorSystems -disable-availability-checking %S/../Inputs/FakeDistributedActorSystems.swift
+// RUN: %target-swift-frontend -typecheck -verify -disable-availability-checking -plugin-path %swift-plugin-dir -parse-as-library -I %t %S/../Inputs/FakeDistributedActorSystems.swift -dump-macro-expansions %s -dump-macro-expansions 2>&1 | %FileCheck %s
+
+import Distributed
+
+@_DistributedProtocol
+protocol Greeter: DistributedActor where ActorSystem == FakeActorSystem {
+  distributed func greet(name: String) -> String
```

Comment:
I'm just going to leave an enhancement request here - I think we need to get tested at least:

- methods with generic parameters
- distributed variables i.e. protocol requirements like `distributed var compute: Int { get set }`
- Multiple distributed members

Reply:
Sounds good, will do

Reply:
Good call on properties actually, I don't think we roundtrip tested them yet

Reply:
Covered all these and will add runtime tests as well

Reply:
Sounds good!